test/storage-luatest/storage_1_1_test.lua

```diff
@@ -531,3 +531,79 @@ test_group.test_noactivity_timeout_for_explicit_master = function(g)
         _G.bucket_gc_wait()
     end, {g.replica_1_a:replicaset_uuid(), bid})
 end
+
+test_group.test_named_config_identification = function(g)
+    t.run_only_if(vutil.feature.persistent_names)
+    local new_cfg_template = table.deepcopy(cfg_template)
+    new_cfg_template.identification_mode = 'name_as_key'
+    new_cfg_template.sharding['replicaset_1'] = new_cfg_template.sharding[1]
+    new_cfg_template.sharding['replicaset_2'] = new_cfg_template.sharding[2]
+    new_cfg_template.sharding[1] = nil
+    new_cfg_template.sharding[2] = nil
+    local new_global_cfg = vtest.config_new(new_cfg_template)
+
+    -- Attempt to configure with named config without name set causes error.
+    g.replica_1_a:exec(function(cfg)
+        ilt.assert_error_msg_contains('Instance name mismatch',
+                                      ivshard.storage.cfg, cfg, 'replica_1_a')
+    end, {new_global_cfg})
+
+    -- Set names on all replicas.
+    g.replica_1_a:exec(function()
+        box.cfg{instance_name = 'replica_1_a', replicaset_name = 'replicaset_1'}
+    end)
+    g.replica_2_a:exec(function()
+        box.cfg{instance_name = 'replica_2_a', replicaset_name = 'replicaset_2'}
+    end)
+
+    -- Check, that UUIDs are validated, when named config is used.
+    local rs_1_cfg = new_global_cfg.sharding.replicaset_1
+    local replica_1_a_cfg = rs_1_cfg.replicas.replica_1_a
+    replica_1_a_cfg.uuid = g.replica_2_a:instance_uuid()
+    g.replica_1_a:exec(function(cfg)
+        ilt.assert_error_msg_contains('Instance UUID mismatch',
+                                      ivshard.storage.cfg, cfg, 'replica_1_a')
+    end, {new_global_cfg})
+    -- The correct UUID should be OK.
+    replica_1_a_cfg.uuid = g.replica_1_a:instance_uuid()
+
+    -- Now the config can be finally applied.
+    vtest.cluster_cfg(g, new_global_cfg)
+
+    -- Test, that sending by name works properly
+    local rs_name_2 = g.replica_2_a:replicaset_name()
+    t.assert_equals(rs_name_2, 'replicaset_2')
+    local bid = g.replica_1_a:exec(function(name)
+        local bid = _G.get_first_bucket()
+        local ok, err = ivshard.storage.bucket_send(
+            bid, name, {timeout = _G.iwait_timeout})
+        ilt.assert_equals(err, nil)
+        ilt.assert(ok)
+        return bid
+    end, {rs_name_2})
+
+    -- Test, that rebalancer is also ok.
+    vtest.cluster_rebalancer_enable(g)
+    g.replica_1_a:exec(function(bid)
+        local internal = ivshard.storage.internal
+        ivtest.service_wait_for_ok(internal.rebalancer_service,
```

Comment:
`service_wait_for_new_ok`. We don't want to accidentally catch the moment when the service was ok because it didn't see the bucket being moved yet.

Reply:
Sure, done